sdk/src/main/scala/coop/rchain/sdk/dag/merging/data/MergeableScope.scala

```diff
@@ -4,57 +4,57 @@ import coop.rchain.sdk.dag.merging.data.MergeableScope.FinalizedDeploys
 
 final case class MergeableScope[B, D](
     latestMsgs: Set[B],
-    finalityFringe: Set[B],
-    finalityState: Map[B, FinalizedDeploys[D]]
-)(implicit bAsBlock: B => Block[B, D]) {
-  // to not do .map(msgSet) all the time
-  implicit val bSetAsBlock: Set[B] => Set[Block[B, D]] = (bs: Set[B]) => bs.map(bAsBlock)
-
+    latestFringe: Set[B],
+    // This index is supposed to be populated on fringe finalization,
+    // therefore it represents all finalized fringes available in local database.
+    finalizationIndex: Map[B, FinalizedDeploys[D]],
```

Comment:
Why would mergeable scope need all available fringes and not just the latest?

Also this info is available in message map in DagMessageState for each message ID.

Reply:
Mergeable scope have to consist of blocks for which indices are available. Its not sufficient to have fringe as a field in `DagMessageState`. This is what I'm trying to express here